lib/register.php

```diff
@@ -517,3 +517,133 @@ function gutenberg_revisions_restore( $revisions_data ) {
 	return $revisions_data;
 }
 add_filter( 'wp_prepare_revision_for_js', 'gutenberg_revisions_restore' );
+
+/**
+ * Generates and caches HTML for a URL that contains OpenGraph metadata.
+ *
+ * @param string $url The URL to generate HTML for.
+ * @return mixed OpenGraph HTML, or false if there was no OpenGraph data.
+ */
+function gutenberg_oembed_opengraph( $url ) {
+	$cache_key = 'gutenberg_opengraph_' . md5( $url );
+	$html      = get_transient( $cache_key );
+
+	// Empty string means we tried to get OpenGraph information for this url, but it had none.
+	if ( '' === $html ) {
+		return false;
+	}
+
+	if ( ! empty( $html ) ) {
+		return $html;
+	}
+
+	$get = wp_safe_remote_get(
+		$url,
+		array(
+			'limit_response_size' => 250000,
+			'user-agent'          => 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:20.0) Gecko/20100101 Firefox/20.0',
```

Comment:
Why is this needed? If there's a good reason, could you leave a comment?

Reply:
Without a user agent, sites like The Verge block requests. Will add a comment!

Reply:
Interesting. Would it be possible to just pass the current user agent? Saves some space and also feels less like cheating haha :) Also this would make is so requests come from a variety of UAs, not just a single one which Gould be detected later?

Reply:
:+1: will do that!